catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/V1WorksController.scala

```diff
@@ -0,0 +1,33 @@
+package uk.ac.wellcome.platform.api.controllers
+
+import com.twitter.inject.annotations.Flag
+import javax.inject.{Inject, Singleton}
+import uk.ac.wellcome.display.models.v1.DisplayWorkV1
+import uk.ac.wellcome.platform.api.services.WorksService
+import uk.ac.wellcome.versions.ApiVersions
+
+@Singleton
+class V1WorksController @Inject()(
+  @Flag("api.prefix") apiPrefix: String,
+  @Flag("api.context.suffix") apiContextSuffix: String,
+  @Flag("api.host") apiHost: String,
+  @Flag("api.scheme") apiScheme: String,
+  @Flag("api.pageSize") defaultPageSize: Int,
+  @Flag("es.index.v1") indexName: String,
+  worksService: WorksService)
+    extends WorksController(
+      apiPrefix = apiPrefix,
+      apiContextSuffix = apiContextSuffix,
+      apiHost = apiHost,
+      apiScheme = apiScheme,
+      indexName = indexName,
+      defaultPageSize = defaultPageSize,
+      worksService = worksService
```

Comment:
Having all this duplication seems a bit icky. Could we use Guice injection to push some of these parameters down into WorksController, or does it not work that way?

Reply:
This is actually already in master as it was merged as part of #1887 (the diff is a bit screwed because this is branched of that one and I think it needs a rebase). ANyway, no Guice can't inject arguments into abstract classes as far as I know cause they can't be instantiatetd